JSDisconnectedException in RadzenDataGrid Dispose

dxrdxr opened the pull request:
As described in #373, in .Net 6.0, there is a race condition where the `JSRuntime` will become unavailable
before a component's `Dispose` is able to be called (may take upwards of 60 seconds). As a result, Blazor will throw a `JSDisconnectedException` on each subsequent call to `JSRuntime.InvokeVoidAsync`.

The fix is to wrap the calls in `Dispose` to catch `JSDisconnectedException`  and set `IsJSRuntimeAvailable` to `false`.
Because `JSRuntime.InvokeVoidAsync` is in fact async, the code must `await` the result in order to catch exceptions.
This pattern of calling into the runtime in `Dispose` is in many components so a wrapper was created to handle the
common boilerplate code needed to wrap the component specific functionality.

The improvement is most notable in the DataGrid component which calls the runtime once per column, which may trigger 4-5 exceptions being thrown per column! After this fix the first exceptions are caught and further exceptions prevented.

This change in behavior is only observed in .Net 6.0, so as part of this fix a new Target was enabled and the associated changes were made.

enchev commented:
Hey @dxrdxr,

Thanks for your pull request however the SASS build is broken after your change. I don't see also where NET5_0_OR_GREATER is defined.

dxrdxr commented:
> 

I see the SASS problem. I think I have an idea what is broken. I look at it later today.

enchev commented:
Maybe we need to restore Before/AfterTargets and the Condition for SASS related targets.

dxrdxr commented:
Fixed. In trying to optimize the multi-target case I created a bug, in fixing that bug I introduced this bug. :-(

In figuring out what went wrong, I noticed that the csproj file is duplicating the `SASS` Target without making any changes other than using `Sass` instead of `SassFile`. By changing the `ItemGroup` to `SassFile` it allows the redundant target to be removed and also enables the `LibSassBuilder` optimizations such as change detection based on a cache of the hashes, which can save 10 seconds off an incremental build by skipping unnecessary Sass builds.

enchev commented:
Unfortunately there is another issue - navigation seems broken:
![image](https://user-images.githubusercontent.com/5804953/154425092-984a1657-ca77-4984-809f-d40f14ab8b12.png)
There are no errors however clicking the navigation menu items and event getting started button will not navigate - just the url in the address bar is changed.

dxrdxr commented:
The problem was a deadlock in the JSRuntime caused by misuse of `await`.

New Fix/PR later today.
